src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt

```diff
@@ -1,17 +1,399 @@
+      ref = prisonService.lookupCodeForReferenceDescriptions(domain, description, false).firstOrNull()
+      if (ref == null) {
+        ref = prisonService.lookupCodeForReferenceDescriptions(domain, description, true).firstOrNull()
+      }
+    }
+    log.debug("Searching for text '{}' in type {} - Found = {}", description, domain, ref)
+    return ref
+  }
+}
+
+data class CourtDataToSync(
+  val courtId: String,
+  val description: String,
+  val longDescription: String? = null,
+  val active: Boolean,
+  val deactivationDate: LocalDate? = null,
+  val addresses: List<AddressDataToSync> = listOf(),
+) {
+
+  override fun hashCode(): Int {
+    var result = courtId.hashCode()
+    result = 31 * result + description.hashCode()
+    result = 31 * result + (longDescription?.hashCode() ?: 0)
+    result = 31 * result + active.hashCode()
+    result = 31 * result + (deactivationDate?.hashCode() ?: 0)
+    return result
+  }
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (javaClass != other?.javaClass) return false
+
+    other as CourtDataToSync
+
+    if (courtId != other.courtId) return false
+    if (description != other.description) return false
+    if (longDescription != other.longDescription) return false
+    if (active != other.active) return false
+    if (deactivationDate != other.deactivationDate) return false
+
+    return true
+  }
+}
+
+data class AddressDataToSync(
+  var addressId: Long? = null,
+  val addressType: ReferenceCode?,
+  val flat: String? = null,
+  val premise: String? = null,
+  val street: String? = null,
+  val locality: String? = null,
+  val town: ReferenceCode? = null,
+  val postalCode: String? = null,
+  val county: ReferenceCode? = null,
+  val country: ReferenceCode? = null,
+  var primary: Boolean,
+  val noFixedAddress: Boolean,
+  var startDate: LocalDate? = null,
+  var endDate: LocalDate? = null,
+  val phones: List<PhoneFromPrisonSystem> = listOf(),
+  var comment: String? = null
+) {
+
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun updateAddressAndPhone(
+    address: AddressDataToSync
+  ) {
+    address.addressId = addressId
+    address.primary = primary
+    address.startDate = startDate
+    address.endDate = endDate
+    address.comment = comment
+    log.debug("MATCH: Court Register address {} and prison court address {}", address, this)
+
+    // update the phones
+    updatePhoneIds(address, this)
+  }
+
+  private fun updatePhoneIds(
+    courtRegAddress: AddressDataToSync,
+    legacyAddress: AddressDataToSync
+  ) {
+    courtRegAddress.phones.forEach { phone ->
+      val matchedPhone = legacyAddress.phones.find { p -> p == phone }
+      if (matchedPhone != null) {
+        phone.phoneId = matchedPhone.phoneId
+        log.debug("MATCH: Court Register phones {} and prison court phones {}", phone, matchedPhone)
+      }
+    }
+  }
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
Why did you have to override these equals/hashCodes in all this data classes?

Reply:
So I could do comparisons on the data to see if it had changed